ansible_base/authentication/common.py

```diff
@@ -101,16 +106,26 @@ def create_claims(authenticator: Authenticator, username: str, attrs: dict, grou
             is_superuser = has_permission
         elif auth_map.map_type == 'is_system_auditor':
             is_system_auditor = has_permission
-        else:
+        elif auth_map.map_type == 'team':
             if auth_map.organization not in org_team_mapping:
                 org_team_mapping[auth_map.organization] = {}
             org_team_mapping[auth_map.organization][auth_map.team] = has_permission
+        elif auth_map.map_type == 'organization':
+            organization_membership[auth_map.organization] = has_permission
+        elif auth_map.map_type == 'role':
+            roles_mapping[auth_map.role] = has_permission
+        else:
+            logger.error(f"Map type {auth_map.map_type} of rule {auth_map.name} does not know how to be processed")
 
     return {
         "access_allowed": access_allowed,
         "is_superuser": is_superuser,
         "is_system_auditor": is_system_auditor,
-        "claims": org_team_mapping,
+        "claims": {
+            "team_membership": org_team_mapping,
+            "roles": roles_mapping,
```

Comment:
I would remove roles here for the following reasons:

1. RBAC is still a work in progress, and I'm not sure if we'll have global roles. Roles may be required to be assigned to a resource such as an Organization.
2. Role associations can be created for the user via their org and team mappings. Users should set up their teams with the level of access that each team requires and then be grouped into those teams at login.

I can see how this would create problems for EDA since doesn't have teams or organizations yet, but I think it makes more sense for them to at least create teams for this feature.